src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -153,22 +171,40 @@ Datum
 create_reference_table(PG_FUNCTION_ARGS)
 {
 	Oid relationId = PG_GETARG_OID(0);
+
+	CreateReferenceTable(relationId);
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * TODO: add comment
+ */
+static void
+CreateReferenceTable(Oid relationId)
+{
+	uint32 colocationId = INVALID_COLOCATION_ID;
+	List *workerNodeList = WorkerNodeList();
 	int shardCount = 1;
-	AttrNumber firstColumnAttrNumber = 1;
+	int replicationFactor = list_length(workerNodeList);
+	Oid distributionColumnType = InvalidOid;
+	char *distributionColumnName = NULL;
 
-	char *firstColumnName = get_attname(relationId, firstColumnAttrNumber);
-	if (firstColumnName == NULL)
+	/* check for existing colocations */
+	colocationId = ColocationId(shardCount, replicationFactor, distributionColumnType);
```

Comment:
> Will we continuously update the replication factor in pg_dist_colocation for reference tables? What if we have a special case, e.g. replication factor 0, or a reserved colocation group ID?

That's a good point. Since we only need to update when a node is added, I'm planning to handle  that #971. Does that sound OK to you?